File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
  m := map[int]int{}
  for i, num := range nums {
    if j, ok := m[target - num]; ok {
      return []int{i, j}
    }
    m[num] = i
  }
  return []int{}
}

File: 0001-two-sum/0001-two-sum.php
class Solution {
  function twoSum(array $nums, int $target): array {
    $m = [];
    foreach ($nums as $i => $num) {
      if (array_key_exists($target - $num, $m)) {
        return [$m[$target - $num], $i];
      }
      $m[$num] = $i;
    }
    return [];
  }
}